### PR TITLE
Cleanup and flake8 fixes for discrete_model

### PR DESCRIPTION
**Pull request by jbrockmendel**
Weibull model had been commented-out in its entirety since at least Oct 2012.  Deleted it.

Removed duplicate and unused imports.  Arranged imports by source

Fixed docstring indentation and some comments in `__main__` that were not indented.  This is useful because it allows for code-folding in some IDEs.

Changed `MultinomialResults._get_endog_name` to avoid built-in name `all` in favor of `use_all`.
Used `pandas.util.decorators.deprecate_kwarg` to keep this backward-compatible.  As a private
method, this should not matter much.

`MultinomialResults._maybe_convert_ynames_int` never uses `self`, so does not need to be a method.  Moved it out to be a private module-level function.

Added shebang.

Left untouched a large number of pep8 complaints of the form:
statsmodels/discrete/discrete_model.py:3096:25: E231 missing whitespace after ','

This PR is very nearly substance-free.  Before long I'll be putting together a substantial MNLogit PR, separating this out to keep the scopes narrow.

### CODE REVIEW

**Comment by josef-pkt** [comment]
this could potentially have a lot of merge conflicts after #3727 is merged

**Comment by jbrockmendel** [comment]
The differences look pretty minor to me, largely because #3727 adds more code than it edits.

The biggest difference is the `__main__` block getting deleted there but not here.  I prefer the straight-out-delete option, so am happy to make that edit.

There may be a whitespace merge conflict in the imports, but nothing non-trivial.

The only genuine conflict I see is that this PR replaces `lm.RegressionResultsWrapper` with importing `RegressionResultsWrapper` directly.  I'd be willing to revert this change in the name of expediency.

**Comment by jbrockmendel** [comment]
Rebased and de-conflicted following merge of #3795

File: statsmodels/discrete/discrete_model.py [hunk 1]
```diff
@@ -324,7 +340,7 @@ def fit_regularized(self, start_params=None, method='l1',
         try:
             kwargs['alpha'] = alpha
         except TypeError:
```
Comment (jbrockmendel) [inline statsmodels/discrete/discrete_model.py]: What would trigger a TypeError here?

File: statsmodels/discrete/discrete_model.py [hunk 2]
```diff
@@ -3442,82 +3393,77 @@ def __init__(self, model, mlefit):
...
+class PoissonResultsWrapper(RegressionResultsWrapper):
     pass
     #_methods = {
     #        "predict_prob" : "rows",
     #        }
-    #_wrap_methods = lm.wrap.union_dicts(
-    #                            lm.RegressionResultsWrapper._wrap_methods,
+    #_wrap_methods = wrap.union_dicts(
+    #                            RegressionResultsWrapper._wrap_methods,
     #                            _methods)
```
Comment (jbrockmendel) [inline statsmodels/discrete/discrete_model.py]: Can we just delete these commented-out _methods and _wrap_methods entirely?

**Comment by josef-pkt** [comment]
I have one last PR for discrete, for now, to merge #3928 then I think this should go in.
I also started line length fixes in test_discrete, but there are too many and they would have messed up the #3928 PR changeset by adding too much noise.

**Comment by josef-pkt** [comment]
I think the Weibull example could be copied to an issue. Deletion is fine, I doubt it will come back to the same module or in closely related form.
I'm not familiar with it, but it might provide some ideas for enhancements.
E.g. the open hurdle count PR uses a censored at 1, binary Poisson which sounds a bit similar, at least superficially.

**Comment by josef-pkt** [comment]
This pull request introduces 1 alert and fixes 15 - [view on lgtm.com](https://lgtm.com/projects/g/statsmodels/statsmodels/rev/pr-4da988e6a871b5363022eab74744b9074523a08e)

**new alerts:**

* 1 for Module-level cyclic import

**fixed alerts:**

* 15 for Module-level cyclic import

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by josef-pkt** [comment]
This pull request introduces 1 alert and fixes 15 - [view on lgtm.com](https://lgtm.com/projects/g/statsmodels/statsmodels/rev/pr-c0a8aa598d8717a52ea48d529cb16005e742f96a)

**new alerts:**

* 1 for Module-level cyclic import

**fixed alerts:**

* 15 for Module-level cyclic import

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by jbrockmendel** [comment]
cleaning up code smells not a priority.  closing.

**Comment by jbrockmendel** [comment]
Fixes 15 alerts... don't know what that means, but it sure sounds like a good thing.

**Pull request closed without merging**